nightly-examples/examples/async_fn.rs

```diff
@@ -0,0 +1,64 @@
+//! Demonstrates using the `trace` attribute macro to instrument `async`
+//! functions.
+//!
+//! This is based on the [`hello_world`] example from `tokio`.
+//!
+//! This server will create a TCP listener, accept connections in a loop, and
+//! write back everything that's read off of each TCP connection.
+//!
+//! Because the Tokio runtime uses a thread pool, each TCP connection is
+//! processed concurrently with all other TCP connections across multiple
+//! threads.
+//!
+//! To see this server in action, you can run this in one terminal:
+//!
+//!     cargo run --example echo
+//!
+//! and in another terminal you can run:
+//!
+//!     cargo run --example connect 127.0.0.1:8080
+//!
+//! Each line you type in to the `connect` terminal should be echo'd back to
+//! you! If you open up multiple terminals running the `connect` example you
+//! should be able to see them all make progress simultaneously.
+//!
+//! [`hello_world`]: https://github.com/tokio-rs/tokio/blob/132e9f1da5965530b63554d7a1c59824c3de4e30/tokio/examples/hello_world.rs
+#![feature(async_await)]
+use tokio;
+use tokio::io::AsyncWriteExt;
+use tokio::net::TcpStream;
+
+use std::error::Error;
+
+#[tracing_proc_macros::trace]
+async fn connect(addr: &std::net::SocketAddr) -> Result<TcpStream, std::io::Error> {
```

Comment:
Lots of full paths, I think the example should better match how you expect people to use it, which would be `info!` instead of `tracing::info!` or `io::Error` instead of `std::io::Error`, etc.

Reply:
I thought it was good to use full paths in the example so that it was clearer where these types are actually coming from when looking at the code. Happy to change it if you think it should be different.

Reply:
I'm somewhat inclined to agree with Jane—the abbreviations are commonly used that full import paths feel off, but this isn't "heck no" review comment, it's more of a "huh, that's interesting".

Reply:
I've changed the example to import more stuff, WDYT?

Reply:
Yeah! looks great to me. I'd just add a comment to the example above this line:

```rust
.with_filter(tracing_fmt::filter::EnvFilter::from("async_fn=trace"))
```
...as to why `async_fn=trace` is needed